### DISCUSSION TITLE
RST pin define for ILI9341 display on ESP32 board with rdagger/micropython-ili9341 library

### DISCUSSION THREAD

**Discussion by paelzer** [micropython/micropython#13565] [Libraries & Drivers]
Hello

I'm trying to get an ILI9341 display attached to an ESP32 board to work with rdagger's ILI9341 library: https://github.com/rdagger/micropython-ili9341

It is this one: https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/blob/main/OriginalDocumentation/2-Specification/ESP32-2432S028%20Specifications-EN.pdf

The reset pin of the **TFT** is attached to the **EN** pin of the ESP32:

![image](https://github.com/micropython/micropython/assets/16374477/cf5d7179-9b4d-45b9-a0ae-1c28d3db23e6)

Problem is since reset pin of TFT isn't connected to a GPIO I don't know what I have to put as rst pin define in the code:

![image](https://github.com/micropython/micropython/assets/16374477/cbdb2579-24bc-41db-8682-ffe43162aa2a)

It doesn't work with `rst=-1` or `rst=None` or also not with different other things I tried like rst=Pin(EN) and so on.

Can you help me and explain how to set it up right if TFT reset pin is not connected to a GPIO pin but to EN or RESET pin?

Thanks a lot and have a nice day!!

Boris

**Comment by karfas** [comment]
> I'm trying to get an ILI9341 display attached to an ESP32 board to work with rdagger's ILI9341 library: https://github.com/rdagger/micropython-ili9341
> Can you help me and explain how to set it up right if TFT reset pin is not connected to a GPIO pin but to EN or RESET pin?

You can't without hardware modification, but you *can* either 
- define any unused GPIO pin as a restart pin for the ILI
- comment or remove any reference to the usage of the rst pin from the linked driver (it's **source**)

The ILI will most likely work without the ability to (hard-)reset it from the program.

**Reply by paelzer** [reply]
Hello

Awesome, thanks a lot. Defining an unused ESP32 pin as "rst" pin, in my case pin 22, works just fine. 

Gruß Boris
